docs/docs/icicle/primitives/merkle.md

```diff
@@ -1,3 +1,270 @@
-# Merkle tree
 
-TODO update for V3
\ No newline at end of file
+# Merkle Tree API Documentation
+
+## What is a Merkle Tree?
+
+A **Merkle tree** is a cryptographic data structure that allows for **efficient verification of data integrity**. It consists of:
+- **Leaf nodes**, each containing a piece of data.
+- **Internal nodes**, which store the **hashes of their child nodes**, leading up to the **root node** (the cryptographic commitment).
+
+---
+
+## Tree Structure and Configuration
+
+### Structure Definition
+
+With ICICLE, you have the **flexibility** to build various tree topologies based on your needs. A tree is defined by:
+
+1. **Hasher per layer** ([Link to Hasher API](./hash.md)) with a **default input size**.
+2. **Size of a leaf element** (in bytes): This defines the **granularity** of the data used for opening proofs.
```

Comment:
What does "granularity" mean here?

Reply:
renamed unit-size. This is the size of a leaf element that is opened with a Merkle proof